scripts/intel_pt_trace.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2024-present ScyllaDB
+# 
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Usage:
+# python scripts/intel_pt_trace.py
+
+# Runs a Scylla cluster, runs some operation (defined in `run()`) on it under Intel PT recording,
+# and outputs out.ftf (Fuchsia trace format) trace to the current working directory.
+
+# Note: needs root rights. Otherwise perf isn't able to record some things,
+# and isn't able to decode some things.
```

Comment:
Does perf_event_paranoid help?

Reply:
Actually, it seems that if you only want to trace userspace, even `perf_event_paranoid` isn't required. (But without `perf_event_paranoid`, unprivileged users have a limit on the mmap size of the trace buffer, which could make gaps in the trace due to overflow more likely). I thought it was much more restrictive. Either I misremembered or the situation has improved since then.

But if you want to trace the kernel, AFAIU you need access to /proc/kcore, and you need higher privileges for that.

Reply:
Yes I think sudo is warranted here (and only perf is run under sudo, not the whole thing).